### PR TITLE
Readds thermobaric and reworks it entirely, new sprite too

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
DS thermobaric untouched, other than it has WP now. Not like it'll matter since if you got hit by it you were dead anyway?
Normal Thermobaric does 40/25/2
Has a WP effect when it hits
50 points and 5 per array. Might increase it to Ten or Seven later.
image

## Why It's Good For The Game
It adds a cool option I guess. It's a cool gun for cool commandos.

## Changelog
:cl:
expansion: added a thermobaric to req, way weaker and cheaper than before.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/magazines/specialist.dm [hunk 1]
```diff
@@ -246,20 +246,31 @@
...
+	default_ammo = /datum/ammo/rocket/wp/quad/ds
+
 /obj/item/ammo_magazine/internal/launcher/rocket/m57a4
 	desc = "The internal tube of an M57A4 thermobaric launcher."
 	caliber = CALIBER_ROCKETARRAY
 	default_ammo = /datum/ammo/rocket/wp/quad
 	max_rounds = 4
 
+/obj/item/ammo_magazine/internal/launcher/rocket/m57a4/ds
+	default_ammo = /datum/ammo/rocket/wp/quad/ds
```
Comment (InterroLouis) [inline code/modules/projectiles/magazines/specialist.dm:272]: Ahem cough cough.

**Comment by Wayland-Smithy** [comment]
> quite nerfed.

> Has a WP effect when it hits

Funny.

**Comment by Pariah919** [comment]
>Funny, since it's basically a 4 shot super remote flamer a la scope plasma pistol meme.
i mean you're spending like 3 miner passes to even buy one so i don't really see the issue with giving it wp effects, splinter asked me to do that else i'd just leave it as is
even then it's just a really bruh incend grenade rn

**Comment by Wayland-Smithy** [comment]
> like 3 miner passes

Big whoop. Just hold the miner closest to the FOB and wait for points to appear automatically over a few min or sell 1 bald fob camping T3. I bet most ops it can be bought before the dropship even lands, also Splinter is a knucklehead sometimes so I'll for sure be asking him how this makes sense too.

**Comment by SplinterGP** [comment]
> 
> 
> > like 3 miner passes
> 
> Big whoop. Just hold the miner closest to the FOB and wait for points to appear automatically over a few min or sell 1 bald fob camping T3. I bet most ops it can be bought before the dropship even lands, also Splinter is a knucklehead sometimes so I'll for sure be asking him how this makes sense too.

you can buy WP grenades which have even bigger range its just a fun weapon idk whats your problem with it.

**Comment by Wayland-Smithy** [comment]
> you can buy WP grenades which have even bigger range its just a fun weapon idk whats your problem with it.

A WP grenade still has a detonation timer, even in a GL (which is too short imo but out of scope for this), and you can't shoot 4 of them in rapid succession. Also am I reading this right 5 points for 4 WP shots? meme?

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -1010,7 +1010,7 @@
 	max_shells = 4 //codex
 	caliber = CALIBER_ROCKETARRAY //codex
 	load_method = MAGAZINE //codex
-	current_mag = /obj/item/ammo_magazine/rocket/m57a4
+	current_mag = /obj/item/ammo_magazine/rocket/m57a4/ds
```
Comment (Wayland-Smithy) [inline code/modules/projectiles/guns/specialist.dm:1013]: defaulting to deathsquad version is just asking for a funny oversight down the road
```suggestion
	current_mag = /obj/item/ammo_magazine/rocket/m57a4
```

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -1167,14 +1167,35 @@ datum/ammo/bullet/revolver/tp44
 	name = "thermobaric rocket"
 	hud_state = "rocket_thermobaric"
 	flags_ammo_behavior = AMMO_ROCKET
-	damage = 200
+	damage = 40
+	penetration = 25
 	max_range = 30
+	sundering = 2
+	var/datum/effect_system/smoke_spread/smoke_system
+	var/danger_message = "<span class='danger'>The rocket explodes explodes into white gas!</span>"
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1175]: No autodoc

Comment (TiviPlus) [reply code/modules/projectiles/ammo_datums.dm:1175]: Also danger_message is unused

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -1167,14 +1167,35 @@ datum/ammo/bullet/revolver/tp44
...
+/datum/ammo/rocket/wp/quad/set_smoke()
+	smoke_system = new /datum/effect_system/smoke_spread/phosphorus()
+
+/datum/ammo/rocket/wp/quad/drop_nade(turf/T, atom/firer, range = 3, radius = 3)
+	set_smoke()
+	smoke_system.set_up(range, T)
+	smoke_system.start()
+	T.visible_message(danger_message)
+	if(!T || !isturf(T))
 		return
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1186]: Duplicate check

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_datums.dm (hunk 3 above)
Comment (Pariah919) [reply code/modules/projectiles/ammo_datums.dm:1175]: danger message works on hit

Comment (TiviPlus) [reply code/modules/projectiles/ammo_datums.dm]: I know it works im saying its useless to have it in a var

**Pull request merged**